### PR TITLE
AGW-3126: Create set-authentication operation to use in custom policies

### PR DESCRIPTION
**Pull request by NachoRaik**

### CODE REVIEW

File: src/main/java/com/mulesoft/extension/policies/authentication/provider/DumbSecurityProvider.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * (or other master license agreement) separately entered into in writing between you and
+ * MuleSoft. If such an agreement is not in place, you may not use the software.
+ */
+package com.mulesoft.extension.policies.authentication.provider;
+
+import org.mule.runtime.api.security.Authentication;
+import org.mule.runtime.api.security.SecurityException;
+import org.mule.runtime.core.api.security.AbstractSecurityProvider;
+
+public class DumbSecurityProvider extends AbstractSecurityProvider {
```
Comment (balbifm) [inline src/main/java/com/mulesoft/extension/policies/authentication/provider/DumbSecurityProvider.java]: Let's change the Dumb for something else

File: src/test/java/com/mulesoft/extension/policies/authentication/AuthenticationExtensionTestCase.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+    DefaultMuleAuthentication auth = (DefaultMuleAuthentication) responseEvent.getAuthentication().get();
+    assertThat(auth.getPrincipal(), is(attributes.getPrincipal()));
+    assertThat(auth.getCredentials(), is(attributes.getCredentials()));
+    assertAuthenticationProperties(auth, attributes);
+  }
+
+  private void assertAuthenticationProperties(DefaultMuleAuthentication auth, AuthenticationAttributes attributes) {
+    if (attributes.getProperties() == null && attributes.getExtraProperties() == null) {
+      assertThat(auth.getProperties() == null, is(true));
```
Comment (balbifm) [inline src/test/java/com/mulesoft/extension/policies/authentication/AuthenticationExtensionTestCase.java]: you have a nullValue()

File: src/test/java/com/mulesoft/extension/policies/authentication/AuthenticationExtensionTestCase.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+//    AuthenticationAttributes attributes = new AuthenticationAttributes(PRINCIPAL, CREDENTIALS)
+//        .setProperties(ImmutableMap.of(PROPERTY_1, VALUE_1, PROPERTY_2, VALUE_2));
+//
+//    flowRunner("set-authentication-flow")
+//        .withAttributes(attributes)
+//        .runExpectingException(instanceOf(SetAuthenticationFailureException.class));
+  }
+
+  @Test
+  public void addPropertiesToAuthentication() throws Exception {
```
Comment (balbifm) [inline src/test/java/com/mulesoft/extension/policies/authentication/AuthenticationExtensionTestCase.java]: What about a test where a property is overriden?

File: src/main/java/com/mulesoft/extension/policies/authentication/AuthenticationOperations.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import javax.inject.Named;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AuthenticationOperations implements Initialisable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AuthenticationOperations.class);
+
+  public static final String AUTHENTICATION_OBJECT_SECURITY_PROVIDER = "authentication-object-security-provider";
```
Comment (balbifm) [inline src/main/java/com/mulesoft/extension/policies/authentication/AuthenticationOperations.java]: gateway-authentication-extension-security-provider

File: pom.xml [hunk 5]
```diff
@@ -8,11 +8,11 @@
         <version>1.1.1</version>
     </parent>
 
-    <artifactId>authentication-object-extension</artifactId>
+    <artifactId>authentication-extension</artifactId>
```
Comment (balbifm) [inline pom.xml]: To follow with conventions: mule-event-authentication-extension

**Comment by NachoRaik** [comment]
Things pending to define:
- Are all dependencies needed? I'll be checking removing one by one, until it fails
- How to test error/failure in the authenticationHandler (Commented now, not sure what to mock)
- The way the properties assertion is being done (not quite satisfied with the way I'm doing it right now)

**Pull request merged**